theano/gpuarray/subtensor.py

```diff
@@ -481,24 +481,46 @@ def c_code_cache_version(self):
         return (0,)
 
 
-class GpuAdvancedSubtensor(HideC, tensor.AdvancedSubtensor):
+def check_and_convert_boolean_masks(input, idx_list):
```

Comment:
Just an idea, if you make an attribute to the op as said in the other comment, mostly, you could have only one implementation of this function in the CPU code that have a parameter to raise or convert.